.github/workflows/ci.yml

```diff
@@ -164,3 +164,26 @@ jobs:
       uses: coverallsapp/github-action@master
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
+  # Make sure the library builds with all dependencies downgraded to their
+  # oldest versions allowed by the semver spec. This ensures we have not
+  # under-specified any dependency
+  minimal-versions:
+    name: Minimal versions
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Install stable Rust
+      uses: actions-rs/toolchain@v1.0.7
+      with:
+        toolchain: stable
+        profile: minimal
+    - name: Install nightly Rust
+      uses: actions-rs/toolchain@v1.0.7
+      with:
+        toolchain: nightly
+        profile: minimal
+    - name: Downgrade dependencies to minimal versions
+      run: cargo +nightly update -Z minimal-versions
+    - name: Compile with minimal versions
+      run: cargo +stable build --all-targets --locked
```

Comment:
- This should be `check`, not `build`
- This should include `--workspace` and `--all-features`
- Ideally the flag order would match other jobs

Reply:
Good feedback. All of it is fixed.